lib/utils/resource-links.js

```diff
@@ -0,0 +1,64 @@
+function getEntries (data) {
+  return (data.items || []).concat((data.includes && data.includes.Entry) || [])
+}
+
+function isContentfulEntryResourceLink (field) {
+  return field && field.sys && field.sys.type === 'ResourceLink' && field.sys.linkType === 'Contentful:Entry'
+}
+
+function getResourceLinkFields (data, locale) {
+  let fields = getEntries(data).flatMap(({ fields }) => Object.values(fields))
+  if (locale === '*') {
+    fields = fields.flatMap((fieldValue) => Object.values(fieldValue))
+  }
+  fields = fields.flat()
+  return fields.filter(isContentfulEntryResourceLink)
+}
+
+function parseResourceLink (resourceLink) {
+  const urn = resourceLink.sys.urn
+  const [, spaceId, , entryId] = urn.split('/')
+  return { spaceId, entryId, urn }
+}
+
+function groupBy (key, objArray) {
+  return objArray.reduce((acc, obj) => {
+    const objKey = obj[key]
+    acc[objKey] = (acc[objKey] || []).concat(obj)
+    return acc
+  }, {})
+}
+
+async function getEntitiesFromSpace (spaceId, entries, extraApis, include, locale) {
+  const apiInstance = extraApis[spaceId]
+  if (!apiInstance) {
+    return []
+  }
+
+  const entryIds = entries.map(({ entryId }) => entryId)
+  const uniqueEntryIds = [...new Set(entryIds)]
+  const response = await apiInstance.getEntries({ 'sys.id[in]': uniqueEntryIds.join(','), include, locale }, { wrapEntryCollection: false }).catch(() => ({}))
+  return getEntries(response)
```

Comment:
I believe this will cause `Asset` fields from the `includes` of the `ResourceLink` object to be discarded

Reply:
I would suggest putting `items` in the `includes.Entry` array and returning the `includes` object as a whole. In `addResourceLinksToIncludes` you can rewrite the merging logic to merge all `includes` keys e.g.
```javascript
const resourceLinkEntries = (await Promise.all(resourceLinkResponses)).flat() // {[type: String]: Record<string, any>[]}[]
const includesObjects = [...resourceLinkEntries, data.includes || {}]
const includesKeys = new Set(includesObjects.flatMap(({includes = {}}) => Object.keys(includes))
return {
  ...data,
  includes: Array.from(...includesKeys).reduce((includes, key) => ({
    ...includes,
    [key]: includesObjects.flatMap((incl) => incl[key] || [])
  }), {})
}
```